src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -528,5 +549,71 @@ public void Dispose()
 
         private string NormalizePath(string path) =>
             path == null ? null : Path.GetFullPath(path.Replace('\\', Path.DirectorySeparatorChar).Replace('/', Path.DirectorySeparatorChar));
+
+        public static string GetLatestVSMSBuildPath()
+        {
+            var programFilesX86 = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86);
+            var programFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+            var windows = Environment.GetFolderPath(Environment.SpecialFolder.Windows);
+
+            var editions = new string[] { "Enterprise", "Professional", "Community" };
+            var targets = new string[] { "Microsoft.CSharp.targets", "Microsoft.CSharp.CurrentVersion.targets", "Microsoft.CSharp.CrossTargeting.targets" };
+
+            foreach (string editionType in editions)
```

Comment:
This logic orders the files by the last time written. So if an older version is installed after a new one it will pick up the older one.
https://github.com/daveaglick/Buildalyzer/blob/da88770a2952f7894c05cd0866ef9d766f78aaea/src/Buildalyzer/Environment/EnvironmentFactory.cs#L174